modules/mono/editor/bindings_generator.cpp

```diff
@@ -3919,57 +3919,67 @@ void BindingsGenerator::_initialize() {
 }
 
 static String generate_all_glue_option = "--generate-mono-glue";
-
-static void handle_cmdline_options(String glue_dir_path) {
-	BindingsGenerator bindings_generator;
-	bindings_generator.set_log_print_enabled(true);
-
-	if (!bindings_generator.is_initialized()) {
-		ERR_PRINT("Failed to initialize the bindings generator");
-		return;
-	}
-
-	CRASH_COND(glue_dir_path.is_empty());
-
-	if (bindings_generator.generate_cs_api(glue_dir_path.path_join(API_SOLUTION_NAME)) != OK) {
-		ERR_PRINT(generate_all_glue_option + ": Failed to generate the C# API.");
-	}
-}
+bool BindingsGenerator::generating_glue;
+String BindingsGenerator::glue_dir_path;
 
 static void cleanup_and_exit_godot() {
 	// Exit once done
 	Main::cleanup(true);
 	::exit(0);
 }
 
-void BindingsGenerator::handle_cmdline_args(const List<String> &p_cmdline_args) {
-	String glue_dir_path;
-
+void BindingsGenerator::process_cmdline(const List<String> &p_cmdline_args) {
 	const List<String>::Element *elem = p_cmdline_args.front();
 
 	while (elem) {
 		if (elem->get() == generate_all_glue_option) {
-			const List<String>::Element *path_elem = elem->next();
+			elem = elem->next();
```

Comment:
This change didn't seem necessary and I think it's now missing a `elem->next()` inside the `if` below.

Reply:
The code prior to this needed to use the `path_elem` variable after the loop was done (which is why it was getting stored), but this refactor doesn't need it anymore, which is why I removed it. Once we're inside this if block, the iteration is done, it will always break out of the loop, so both this assigment to a separate variable and the `elem-next()` below aren't needed anymore, that's why it ended up like this.

Reply:
> Once we're inside this if block, the iteration is done, it will always break out of the loop

You are right, I think I didn't this before.